openedx/features/genplus_features/genplus/api/v1/serializers.py

```diff
@@ -86,11 +87,28 @@ class ClassListSerializer(serializers.ModelSerializer):
     current_unit = serializers.SerializerMethodField('get_current_unit')
     lesson = serializers.SerializerMethodField('get_lesson')
 
+    def lastest_unlocked_unit(self, gen_class):
+        # Returns: Latest unlocked unit
+        reversed_class_units = gen_class.class_units.all().reverse()
+
+        for class_unit in reversed_class_units:
+            if not class_unit.is_locked:
+                return class_unit
+        return None
```

Comment:
Can we create a django query to fetch the last unlocked unit?

Reply:
Query would works on attributes, is_locked not an attribute in case of units.